controller/volume_controller.go

```diff
@@ -801,16 +805,21 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, es map[stri
 			return err
 		}
 		if scheduledReplica == nil {
-			logrus.Errorf("unable to schedule replica %v of volume %v", r.Name, v.Name)
-			v.Status.Conditions = types.SetCondition(v.Status.Conditions,
-				types.VolumeConditionTypeScheduled, types.ConditionStatusFalse,
-				types.VolumeConditionReasonReplicaSchedulingFailure, "")
+			if r.Spec.HardNodeAffinity == "" {
+				logrus.Errorf("unable to schedule replica %v of volume %v", r.Name, v.Name)
+				v.Status.Conditions = types.SetCondition(v.Status.Conditions,
+					types.VolumeConditionTypeScheduled, types.ConditionStatusFalse,
+					types.VolumeConditionReasonReplicaSchedulingFailure, "")
+			} else {
+				logrus.Errorf("unable to schedule replica %v of volume %v with HardNodeAffinity = %v", r.Name, v.Name, r.Spec.HardNodeAffinity)
+				v.Status.Conditions = types.SetCondition(v.Status.Conditions,
+					types.VolumeConditionTypeScheduled, types.ConditionStatusFalse,
+					types.VolumeConditionReasonLocalReplicaSchedulingFailure, "")
```

Comment:
I update volume's condition to notify user when Longhorn fail to keep a local replica.